pydantic_ai_slim/pydantic_ai/models/__init__.py

```diff
@@ -506,15 +509,49 @@ def cached_async_http_client(*, provider: str | None = None, timeout: int = 600,
 @cache
 def _cached_async_http_client(provider: str | None, timeout: int = 600, connect: int = 5) -> httpx.AsyncClient:
     return httpx.AsyncClient(
-        transport=_cached_async_http_transport(),
+        transport=_get_transport_for_loop(),
         timeout=httpx.Timeout(timeout=timeout, connect=connect),
         headers={'User-Agent': get_user_agent()},
     )
 
 
 @cache
-def _cached_async_http_transport() -> httpx.AsyncHTTPTransport:
-    return httpx.AsyncHTTPTransport()
+def _get_transport_for_loop() -> _PerLoopTransport:
+    return _PerLoopTransport()
+
+
+# transport._pool._connections[0]._connection._network_stream._stream.transport_stream._transport._loop
+class _PerLoopTransport(httpx.AsyncBaseTransport):
+    def __init__(self):
+        self.transports: MutableMapping[asyncio.AbstractEventLoop, httpx.AsyncHTTPTransport] = (
+            weakref.WeakKeyDictionary()
+        )
+
+    def get_transport(self) -> httpx.AsyncHTTPTransport:
+        # Clean the dictionary of closed loops
+        for loop in list(self.transports.keys()):
+            if loop.is_closed():
+                del self.transports[loop]
+
+        return self.transports.setdefault(asyncio.get_running_loop(), httpx.AsyncHTTPTransport())
```

Comment:
Because I'm using setdefault here, it means `httpx.AsyncHTTPTransport()` is always assigned even when it's not going to be used. Is that something we want to optimize ?